### DISCUSSION TITLE
Mapping vs Matching - what is what?

### DISCUSSION THREAD

**Discussion by matentzn** [mapping-commons/sssom#169] [General]
In an offline discussion with @graybeal and @cpesquita we were looking at trying to at least roughly agree on what we mean my `mapping` and what we mean by `matching`. My [related tweet](https://twitter.com/NicoMatentzoglu/status/1517186240050831363) revealed a lot of different ways to put it, but the overarching theme to me seems that the string "mapping" typically refers to a tuple, a "continuant" if you will, while "matching" is a process which may or may not result in a tuple. 

Obviously there also _is_ an activity called "mapping", and here is where my problem starts. According to some of the respondents on twitter and @cpesquita the difference between "mapping" and "matching" is in practice a bit unclear. Some people were emphasising the "automated" aspect of "matching". What I want to know is: 

1. first, do you vaguely agree with the fact that "mapping" typically refers to a tuple, a "continuant" if you will, while "matching" is a process which may or may not result in a tuple? 
2. How should we refer to a non-automated activity of identifying a correspondence between two terms, like a person that "maps" a term "A" in one ontology to a term "B" in another, by applying a range of complex searches, and judgements based on domain knowledge. Would you still call that process of identifying the correspondence "matching"? Intuitively it seems that person is "mapping", but happy to defer to the experts. We need to know an answer to this so we can add these kinds of activities to our vocabulary to describe activities involved in the creation of mappings.

**Comment by KrishnaTO** [comment]
Not an expert opinion, but my 2 bits:
1. I personally don't equate mapping to be restricted to a tuple, while matching, with its number of meaning across domains, should be an ontological slang.
2.  Mapping should be used for when an association is formally made, and can be referenced. 
> Would you still call that process of identifying the correspondence "matching", [a person that "maps" a term "A" in one ontology to a term "B" in another, by applying a range of complex searches, and judgements based on domain knowledge]?

As long as it's formally captured and can be referenced, yes. I think of it like, those entities 'matched' using that pattern, but the same pattern may not lead to the same 'match' unless the metadata can be replicated (especially the versions for all involved). Having 'mapped' a term ought to be a controlled and precisely equivalency. 
Not that it's strictly relevant but, especially when considering that a term no longer remains existentially contained within a single domain, but refers to more than a single domain, just requires it be stronger than just being a 'match'.

**Reply by matentzn** [reply]
Thank you @KrishnaTO, most of this makes sense! What do you mean by "no longer remains existentially contained within a single domain"?

**Reply by KrishnaTO** [reply]
Single domain (in this context) means a single ontology or subject topic, for my view.

**Reply by matentzn** [reply]
Got it! Thanks.

**Reply by jonquet** [reply]
I personally use the following expressions :
- ontology matching is the process of aligning two ontologies. 
- ontology alignment is the result of the process. An alignment being a set of mappings between ontology entities. 
- an ontology mapping or more correctly à concept mapping is an element in an alignment set. 

Sometime ontology matching or alignment are similarly used to denote the research domain.

**Reply by matentzn** [reply]
Interesting @jonquet this makes total sense to me. I have a slightly more specialised notion of alignment (it should be to some degree "complete", i.e. I wouldn't call a mapping set with a single mapping between DO and SNOMED an "alignment"). Therefore matching is also slightly more general, as it could result in a partial alignment. But no serious conceptual differences here I think.

**Comment by matentzn** [comment]
OHDSI says: https://www.ohdsi.org/web/wiki/doku.php?id=documentation:vocabulary:mapping

- Mapping, also known as a map, is an association between a particular concept in one code system or dataset and code in another, rarely the same, code system that has the same (or similar) meaning.
- Also, Mapping is the process to transform one concept into a Standard one. The Standardized Clinical Data Tables allow only Standard Concepts, thus all other codes used in the source databases have to be translated to Standard Concepts.

I think its fine to use it for both, but I just want to be clear about the fact that "people are performing a matching activity which results in a mapping" is a sentence that makes some sense.

**Reply by graybeal** [reply]
I kind of liked Clement's definitions. But then I got to thinking…

Isn't it 'great' that in English the same words can be both a process, and the product of that process?  So a process 'ontology mapping' (mapping ontologies) can produce an ontology mapping, a process of ontology alignment can produce an ontology alignment, and a process of concept mapping can produce one—well, or more—concept mappings.  It doesn't _quite_ work with 'ontology matching' though—there the noun is invariably 'ontology match', and implies matching the whole ontology to another whole ontology (but usually in my experience to mean an exact match, as in they are the same ontology). 

And if I take the 3 or 4 definitions of those 3 or 4 things and switch them around so they are stuck on one of the other terms, no one can argue with any rigor that the new pairings are wrong, because we all use different terms to mean the same thing/the same terms to mean different things!

It (English) is so sad that way…  It's why we have to have a key, or glossary, or dictionary, at the top of our publication, where we describe exactly what these terms mean. And not by re-using the words that name the terms—they have to be precise definitions using (relatively) unambiguous language. (Good luck with that!)

So in response to your question, (1) I can't argue with that being 'vaguely' correct. A mapping is a phrase often used to describe a result that contains two concepts and a relationship between them. And 'matching' is often a process. So, yeah, close-ish.  (2) If I'm writing a paper, I would call `a non-automated activity of identifying a correspondence between two terms` a "manual pair-wise alignment identification process", because that is the shortest expression that captures the 5 things you specified (non-automated; between two terms; correspondence; identification; activity). 

Trying to make single words correspond to ideas of that complexity is highly unlikely to improve the reader's understanding of what you are trying to say. And even more so if the single words have been defined dozens of different ways over the years. Making up a pseudo-word (MPAIP!) to represent your definition is better, at least it isn't starting out life with a confused history.

**Reply by matentzn** [reply]
hahah mpaip.. I agree with all you say, I just would prefer to use the world "alignment" to mean something more holistic. I think you align ontologies, not terms.. and an alignment should have some (any) degree of completeness to it to be called "alignment".

The easiest way out of this terminological jungle is to consider `a non-automated activity of identifying a correspondence between two terms` as some kind of matching - that way we do not have to climb any mountains when building our controlled vocabulary. If we make this very clear in our prose (tutorials, papers etc) do you think it will cause confusion?

**Reply by graybeal** [reply]
I think you can make it clear by defining your words, but it is much harder to detect the places you (or some other writer) used 'matching' in a case that was automated. Or was a noun (result) rather than a process (activity). Or was matching three terms. Then everyone says "Wait, I thought a matching was defined differently up there..."  It just isn't clear to me why we're putting so much emphasis on the non-automated bit. Does it really change anything if it's non-automated? Manual doesn't necessarily mean 'expert' or 'correct'.

**Reply by matentzn** [reply]
Yeah, I merely suggested this distinction as I feel that `matching` in general parlance is "automated". From your comments throughout I gather you would favour ditching the notion of `matching` throughout in favour of using `mapping` for everything (the tuple, the process)?

**Comment by cpesquita** [comment]
Dear all,

Just a few thoughts:
- I agree that it is unnecessary to distinguish whether a matching
(process) is automated or manual by using a different term for it. Those
would be subclasses of "matching".

- I also use "alignment" in a holistic way: the full set of mappings
(tuples) found between two ontologies by a matching process.

Perhaps we could run a survey about the uses of these terms by researchers?
It would be quite easy to reach out the Ontology Matching community, if we
wanted to get a broader view of what is out there.

Best,
Catia

**Reply by matentzn** [reply]
Survey is a great idea! Thanks @cpesquita
